crates/era-cheatcodes/src/cheatcodes.rs

```diff
@@ -1336,3 +1467,42 @@ fn get_calldata<H: HistoryMode>(state: &VmLocalStateData<'_>, memory: &SimpleMem
         fat_data_pointer.length as usize,
     )
 }
+
+fn compare_logs(expected_logs: &[LogEntry], actual_logs: &[LogEntry], checks: EmitChecks) -> bool {
+    let mut expected_iter = expected_logs.iter().peekable();
+    let mut actual_iter = actual_logs.iter();
+
+    while let Some(expected_log) = expected_iter.peek() {
+        if let Some(actual_log) = actual_iter.next() {
+            if are_logs_equal(expected_log, actual_log, &checks) {
+                expected_iter.next(); // Move to the next expected log
+            } else {
+                return false
+            }
+        } else {
+            // No more actual logs to compare
+            return false
+        }
+    }
+
+    true
+}
+
+fn are_logs_equal(a: &LogEntry, b: &LogEntry, emit_checks: &EmitChecks) -> bool {
+    let address_match = match emit_checks.address {
+        Some(address) => b.address == address,
+        None => true,
+    };
+
+    let topics_match = emit_checks.topics.iter().enumerate().all(|(i, &check)| {
+        if check {
+            a.topics.get(i) == b.topics.get(i)
+        } else {
+            true
+        }
+    });
+
+    let data_match = if emit_checks.data { a.data == b.data } else { true };
+
+    address_match && topics_match && data_match
+}
```

Comment:
Pretty sure there's a simpler way to assert logs, but let's leave that for a round of refactors at a later date :)